### PR TITLE
feat(url): set sort values and read them from the url.

### PR DESCRIPTION
**Pull request by guillei10**
# Description

The goal of this PR is to add the sort to the url flow. Now we will be able to load the sort from the url and also the url is being updated with the values of the sort. Things done:

- Removed the `value` prop from the `Sort` _mixin_. When used it caused duplicated requests.
- Added mutation and wiring in the url module to be updated when the sort changes.
- Changed the action in search module that allows to update the search state with the values coming from the url

### CODE REVIEW

File: packages/x-components/src/x-modules/search/components/__tests__/sort-dropdown.spec.ts [hunk 1]
```diff
@@ -23,7 +23,6 @@ function renderSortDropdown({
       </template>
    </SortDropdown>`,
   items = ['Relevance', 'Price low to high', 'Price high to low'],
```
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/search/components/__tests__/sort-dropdown.spec.ts:25]: Given that we agreed to have the default value be `''`, should we replace `'Relevance'` here with `''`?

File: packages/x-components/src/x-modules/search/components/__tests__/sort-list.spec.ts [hunk 2]
```diff
@@ -20,7 +20,6 @@ function renderSortList({
       </template>
    </SortList>`,
   items = ['Relevance', 'Price low to high', 'Price high to low'],
```
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/search/components/__tests__/sort-list.spec.ts:22]: Same as before, should we replace `''Relevance` with `''`?

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/search/components/__tests__/sort-dropdown.spec.ts [hunk 3]
```diff
@@ -130,71 +123,24 @@ describe('testing SortDropdown component', () => {
...
-      eventPayload: 'price asc',
-      metadata: { moduleName: 'search' }
-    });
-  });
-
   // eslint-disable-next-line max-len
   it('emits the first element of the `items` prop as the provided sort if no `value` is provided', () => {
     const { onSelectedSortProvided } = renderSortDropdown({
-      items: ['price desc', 'price asc', 'default']
+      items: ['price desc', 'price asc', '']
```
[the remark this answers is not in the document]
Comment (guillei10) [reply packages/x-components/src/x-modules/search/components/__tests__/sort-dropdown.spec.ts:129]: I just changed the items to make it consistent and make one of the items to be `''` as it is the default value. I'm changing the description.

File: packages/x-components/src/x-modules/search/components/sort.mixin.ts [hunk 4]
```diff
@@ -50,6 +40,6 @@ export default class SortMixin extends Vue {
    */
   @XEmit('SelectedSortProvided')
   public get providedSelectedSort(): Sort {
-    return this.value ?? this.items[0];
+    return this.selectedSort ?? this.items[0];
```
[the remark this answers is not in the document]
Comment (guillei10) [reply packages/x-components/src/x-modules/search/components/sort.mixin.ts]: I think so.

**Comment by tiborux** [comment]
Should we document that the default value will be always ''? I mean, if you are sending items=['1','2','3'] the component will render also ' ' because its the default value so I dont know if we have to document that the user have to add '' as the first item or warn him that he will see '' as default value.

File: packages/x-components/src/x-modules/search/components/__tests__/sort-dropdown.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (CachedaCodes) [reply packages/x-components/src/x-modules/search/components/__tests__/sort-dropdown.spec.ts:129]: As we just discussed, it's weird having `''` as the default value, even knowing that the adapter will treat this value as sorting by relevance we would have kind of hardcode it in the front to represent that in the sort dropdown or whatever we're using for sorting. 

Setting `items[0]` as the default value for the `selectedSort` or a good way of representing the default value to the customer would be better.

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

**Review by CachedaCodes — DISMISSED** [review]

**Review by tiborux — CHANGES_REQUESTED** [review]

**Review by CachedaCodes — APPROVED** [review]

**Review by tiborux — APPROVED** [review]

**Pull request merged**